gstd/gstd_http.c

```diff
@@ -0,0 +1,423 @@
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  address = soup_message_get_uri (msg);
+
+  message = g_strdup_printf ("read %s", soup_uri_get_path (address));
+  ret = gstd_parser_parse_cmd (session, message, output);       // in the parser
+  g_free (message);
+
+  return ret;
+}
+
+static GstdReturnCode
+do_post (SoupServer * server, SoupMessage * msg, char *name,
+    char *description, char **output, GstdSession * session)
+{
+  gchar *message = NULL;
+  SoupURI *address = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  address = soup_message_get_uri (msg);
+
+  message = g_strdup_printf
+      ("create %s %s %s", soup_uri_get_path (address), name, description);
+  ret = gstd_parser_parse_cmd (session, message, output);       // in the parser
+  g_free (message);
+
+  return ret;
+}
+
+static GstdReturnCode
+do_put (SoupServer * server, SoupMessage * msg, char *name, char **output,
+    GstdSession * session)
+{
+  gchar *message = NULL;
+  SoupURI *address = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  address = soup_message_get_uri (msg);
+
+  message = g_strdup_printf ("update %s %s", soup_uri_get_path (address), name);
+  ret = gstd_parser_parse_cmd (session, message, output);       // in the parser
+  g_free (message);
+
+  return ret;
+}
+
+static GstdReturnCode
+do_delete (SoupServer * server, SoupMessage * msg, char *name,
+    char **output, GstdSession * session)
+{
+  gchar *message = NULL;
+  SoupURI *address = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (server, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (msg, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (session, GSTD_NULL_ARGUMENT);
+
+  address = soup_message_get_uri (msg);
+
+  message = g_strdup_printf ("delete %s %s", soup_uri_get_path (address), name);
+  ret = gstd_parser_parse_cmd (session, message, output);       // in the parser
+  g_free (message);
+  return ret;
+}
+
+static void
+do_request (SoupServer * server, SoupMessage * msg, GHashTable * query,
+    GstdSession * session, verb request_verb)
+{
+  gchar *response = NULL;
+  gchar *name = NULL;
+  gchar *description_pipe = NULL;
+  SoupURI *address = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+  gchar *output = NULL;
+  const gchar *description = NULL;
+  SoupStatus status = SOUP_STATUS_OK;
+  const gchar *query_text = NULL;
+
+  g_return_if_fail (server);
+  g_return_if_fail (msg);
+  g_return_if_fail (session);
+
+  address = soup_message_get_uri (msg);
+
+  query_text = soup_uri_get_query (address);
+  if (request_verb != GET) {
+
+    if (!query_text) {
```

Comment:
This is kind of hard to read and main have get worst when we extend it (for example POST supporting both body and query). I recommend always getting name and description and passing them to the functions and letting them decide if they need them or not.